domenetjenester/uttak/src/main/java/no/nav/foreldrepenger/domene/uttak/uttaksgrunnlag/fp/MorsJustering.java

```diff
@@ -60,6 +67,55 @@ class MorsJustering implements ForelderFødselJustering {
         return fjernHullPerioder(justertePerioder);
     }
 
+    private int antallDagerJustertInnIOppholdSomErForbeholdtMorEtterFødsel(OppgittPeriodeEntitet oppgittPeriode, List<OppgittPeriodeEntitet> ikkeFlyttbarePerioder) {
+        return oppgittPeriodeSomOverlapperMedHullForbeholdMorEtterFødsel(oppgittPeriode, ikkeFlyttbarePerioder).stream()
+            .mapToInt(seg -> beregnAntallVirkedager(seg.getFom(), seg.getTom()))
+            .sum();
+    }
+
+    private static int antallDagerFellesperiodeErRedusert(List<OppgittPeriodeEntitet> justert, List<OppgittPeriodeEntitet> justertOgRedusert) {
+        var antallVirkedagerEtterJustering = justert.stream().mapToInt(p -> beregnAntallVirkedager(p.getFom(), p.getTom())).sum();
+        var antallVirkedagerEtterReduksjon = justertOgRedusert.stream().mapToInt(p -> beregnAntallVirkedager(p.getFom(), p.getTom())).sum();
+        return antallVirkedagerEtterJustering - antallVirkedagerEtterReduksjon;
+    }
+
+
+    /**
+     * Denne klassen reduserer bare fellesperiode som ligger etter ny TOM dato forbeholdt mor ifm fødsel
+     */
+    private List<OppgittPeriodeEntitet> reduserJusterteFellesperioder(List<OppgittPeriodeEntitet> justerteFellesperioder, int antallDagerSomKanReduseres) {
+        List<OppgittPeriodeEntitet> justertFellesperiode = new ArrayList<>();
+        var justerbarPeriode = new LocalDateTimeline<>(TidsperiodeForbeholdtMor.tilOgMed(nyFamiliehendelse).plusDays(1), Tid.TIDENES_ENDE, null);
+        var fellesperioder = new LocalDateTimeline<>(justerteFellesperioder.stream()
+            .map(p -> new LocalDateSegment<>(p.getFom(), p.getTom(), p))
+            .toList());
+        var fellsperiodeSomKanReduseres = fellesperioder.intersection(justerbarPeriode).stream()
+            .map(LocalDateSegment::getValue)
+            .sorted(Comparator.comparing(OppgittPeriodeEntitet::getFom))
+            .toList();
+
+        // Starter fra siste periode til første for å fjerne/justere perioder fra høyre til venstre
+        for (var periode : fellsperiodeSomKanReduseres.reversed()) {
+            var antallVirkedagerForPeriode = Virkedager.beregnAntallVirkedager(periode.getFom(), periode.getTom());
+            if (antallDagerSomKanReduseres == 0) {  // Vi er ferdig justert. Legg til resten av periodene
+                justertFellesperiode.add(periode);
+            } else if (antallVirkedagerForPeriode > antallDagerSomKanReduseres) { // En periode spiser opp resten av forskyvningen
+                var nyTom = Virkedager.plusVirkedager(periode.getFom(), antallVirkedagerForPeriode - antallDagerSomKanReduseres - 1); // -1 fordi virkedager periode.getFom() ikke er telt med
+                justertFellesperiode.add(kopier(periode, periode.getFom(), nyTom));
```

Comment:
Er det mer effektivt å beholde TOM siden du går fra høyre og gobbler dager? Da kan kanskje perioden til venstre for denne shiftes mot høyre uten å også måtte shifte den nye kopien man lager her?

Reply:
Rent praktisk vil vi havne inn i denne if-løkken bare en gang fordi forskyvningen spises opp av fellesperioden som sjekkes på. Så etter dette skal vi beholde resten av fellesperiodene slik som de var justert. 

Så de periodene vi sjekker på her er ferdig justert (hull og forskyvning forbi hull hensyntatt). Så her er det ingenting som trenger å shiftes mot høyre?

Reply:
Siden den kalles kun en gang så kunne du sett på å sluke evt fellesperiode før for (oppgitt : oppgitte-spiste)

Reply:
Hvis du beholder "nyFom...tom" så vil vi avkorte perioden fra venstre av og da vil det oppstå hull som ellers ikke var hull. Hvis vi beholder "fom...nyTom" så avkorter vi fra høyre av og det høres mer riktig ut i mine øre ettersom vi allerede har skjøvet denne perioden til høyre (vi avkorter aldri mer enn forskyvning her).

Reply:
La oss få det ut

Reply:
@espenwaaga når du avkorter fra høyre, så må vel alle senere perioder flyttes mot venstre - et gjør du sikkert
Det jeg lurte på var om det gikk an å gjøre avkortingen før den løkka som dytter mot høyre